File: src/project.c
/*
 * XENLY - high-level and general-purpose programming language
 * created, designed, and developed by Cyril John Magayaga (cjmagayaga957@gmail.com, cyrilmagayaga@proton.me).
 *
 * It is initially written in C programming language.
 *
 */
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include "color.h"

// Initialize project
void initialize_project() {
    // Create a new folder for the project
    system("mkdir xenly_project");

    // Change directory to the newly created folder
    chdir("xenly_project");

    // Create a new Xenly source file
    FILE *source_file = fopen("main.xe", "w");
    if (source_file == NULL) {
        perror("Unable to create source file");
    }

    // Write default "hello world" program to the source file
    fprintf(source_file, "print(\"Hello, World!\")\n\nprint(2*9-6/3*5)\n");
    fclose(source_file);
    
    // Inform the user that the project has been initialized
    printf("New Xenly project initialized in ");
    setBackgroundGreen();
    white();
    printf(" 'xenly_project' ");
    resetColor();
    resetBackgroundColor();
    printf(" folder.\n");
}

// Create initialize project
void create_initialize_project(const char *project_name) {
    // Create a new folder for the project
    char mkdir_command[100];
    sprintf(mkdir_command, "mkdir %s", project_name);
    if (system(mkdir_command) != 0) {
        perror("Unable to create project directory");
        return;
    }

    // Change directory to the newly created folder
    if (chdir(project_name) != 0) {
        perror("Unable to change directory to project folder");
        return;
    }

    // Create a new Xenly source file
    FILE *source_file = fopen("main.xe", "w");
    if (source_file == NULL) {
        perror("Unable to create source file");
        return;
    }

    // Write default "hello world" program to the source file
    fprintf(source_file, "print(\"Hello, World!\")\n\nprint(2*9-6/3*5)\n");

    // Close the source file
    fclose(source_file);

    // Inform the user that the project has been initialized
    printf("New Xenly project initialized in ");
    setBackgroundGreen();
    white();
    printf(" '%s' ", project_name);
    resetColor();
    resetBackgroundColor();
    printf(" folder.\n");
}

File: src/project.rs
/*
 * XENLY - high-level and general-purpose programming language
 * created, designed, and developed by Cyril John Magayaga (cjmagayaga957@gmail.com, cyrilmagayaga@proton.me).
 *
 * It is initially written in Rust programming language.
 *
 * It is available for Linux and Windows operating systems.
 *
 */

use std::fs;
use std::io::{self, Write};
use std::path::Path;

pub fn create_initialize_project(project_name: &str) {
    let project_path = Path::new(project_name);

    if project_path.exists() {
        println!("Error: Project {} already exists.", project_name);
        return;
    }

    if let Err(e) = fs::create_dir(project_path) {
        println!("Error: Could not create project directory {}: {}", project_name, e);
        return;
    }

    create_file(project_path, "main.xe", "nota(\"Hello, World!\")\n\nnota(2*9-6/3*5)\n");

    println!("Project {} created successfully.", project_name);
}

pub fn initialize_project() {
    println!("Initializing a new Xenly project...");

    let mut project_name = String::new();
    print!("Enter project name: ");
    io::stdout().flush().unwrap();
    io::stdin().read_line(&mut project_name).unwrap();
    let project_name = project_name.trim();

    create_initialize_project(project_name);
}

fn create_file(project_path: &Path, file_name: &str, content: &str) {
    let file_path = project_path.join(file_name);
    let mut file = fs::File::create(file_path).expect("Error: Could not create file");
    file.write_all(content.as_bytes()).expect("Error: Could not write to file");
}
